### PR TITLE
Add new Stacks theme

### PR DESCRIPTION
**Pull request by scruffian**
This adds a new theme which can be used to create presentation decks. This idea is that the stacks are supplied in a pattern, which a user can add to a post or page and then use each post/page as a slide deck. The home page creates a demo which explains how it works.

The name "Stack" is already taken but "Stacks" seems to be free.

Design by @SaxonF

### CODE REVIEW

File: stacks/readme.txt
Comment (MaggieCabrera) [inline stacks/readme.txt]: According to [the guidelines](https://make.wordpress.org/themes/handbook/review/required/#9-files), this should be just one wp.org username, so maybe `wordpressdotorg` is best here. I'm not sure where is the best place to name the contributors. The [guidelines](https://make.wordpress.org/themes/handbook/review/required/#9-files) say 

> Themes can include one single front-facing credit link, which is restricted to the Theme URI or Author URI defined in style.css.

I think we should discuss what's the best way to do this and add that info to the readme file on this repo so we are consistent on all our themes

File: stacks/style.css
Comment (MaggieCabrera) [inline stacks/style.css]: Maybe this is where we add the contributor's names? but it might be very long for some themes and this is what it shows here:

Screenshot 2023-03-09 at 16 37 00

File: stacks/style.css
Comment (MaggieCabrera) [inline stacks/style.css]: This is different from what readme.txt says. Out of curiosity, is this coming from CBT?

File: stacks/style.css
Comment (MaggieCabrera) [inline stacks/style.css]: if it's not a child theme I think this can go away. Again, if this is from CBT, we should maybe file an issue so that it's not included when we are not creating a child theme

File: stacks/patterns/index.php
Comment (MaggieCabrera) [inline stacks/patterns/index.php]: We should make all the strings translatable. I think @madhusudhand had a script for that somewhere? Could that be in a package we can use in this repo?

File: stacks/patterns/index.php
Comment (MaggieCabrera) [inline stacks/patterns/index.php]: Since the background colors of the slides are not part of the theme's palette, should we use something other than gray? this is a naive question from a non designer perspective, I'd just think it would look cooler. In the same way you added the image to the last slide, that made the whole theme much more solid, in my opinion. Would love other designer's eyes on this for feedback.

File: stacks/readme.txt
Comment (MaggieCabrera) [reply stacks/readme.txt]: I opened [this PR](https://github.com/WordPress/community-themes/pull/14) for discussion

File: stacks/style.css
Comment (scruffian) [reply stacks/style.css]: Yeah I think so

Comment (MaggieCabrera) [reply stacks/style.css]: Let's make it 6.1 or 6.2, since we need the sticky positioning

File: stacks/patterns/index.php
Comment (jffng) [inline stacks/patterns/index.php]: Should this be called `stacks/home` and be named `home.php` instead, since it's used in the home template?

File: stacks/patterns/index.php
Comment (jffng) [reply stacks/patterns/index.php]: That function lives [here](https://github.com/Automattic/themes/blob/trunk/theme-utils.mjs#L1213) and is used as an [npm script](https://github.com/Automattic/themes/blob/trunk/package.json#L31)

Maybe we could start a paired down version of `theme-utils` in this repo, and eventually extract the shared utilities into an npm package?

File: stacks/patterns/index.php
Comment (jffng) [inline stacks/patterns/index.php]: I noticed this text is black in the site editor, but white on the front end. Any idea why? 

Screenshot 2023-03-09 at 10 32 04 PM

Also it shows up with padding but maybe that's a bigger site wide padding issue. 

Running Gutenberg trunk.

**Comment by MaggieCabrera** [comment]
I wonder if a nav block with anchor links to each section would work here. Maybe adding a small script to smooth the scroll between jumps? When I have a moment I'm going to test that

**Comment by carolinan** [comment]
"Stacks" doesnt relate to presentations or presentation decks to me. Is there any other name that would make it more intuitive to search for the theme's purpose?

**Comment by scruffian** [comment]
> Is there any other name that would make it more intuitive to search for the theme's purpose?

Deck and pitch are unavailable, but slide(s) might be available.

**Comment by scruffian** [comment]
> I wonder if a nav block with anchor links to each section would work here.

Where would it sit? We don't have many controls for z-index so I think it would get hidden behind the panels...

File: stacks/patterns/index.php
Comment (scruffian) [reply stacks/patterns/index.php]: Done

File: stacks/patterns/index.php
Comment (scruffian) [reply stacks/patterns/index.php]: I made a script for this here: https://github.com/WordPress/community-themes/pull/20

**Comment by MaggieCabrera** [comment]
> > I wonder if a nav block with anchor links to each section would work here.
> 
> Where would it sit? We don't have many controls for z-index so I think it would get hidden behind the panels...

yeah, I gave it a try and it didn't work, we need a fixed position now!!!

File: stacks/style.css
Comment (scruffian) [reply stacks/style.css]: Done

File: stacks/style.css
Comment (scruffian) [reply stacks/style.css]: Done

File: stacks/patterns/index.php
Comment (scruffian) [reply stacks/patterns/index.php]: Updated the strings

File: stacks/patterns/index.php
Comment (scruffian) [reply stacks/patterns/index.php]: This is a Gutenberg issue I am investigating

**Comment by scruffian** [comment]
Still outstanding on this are:
1. Should the slides use different colors?
2. What name should we go with?

**Comment by MaggieCabrera** [comment]
> Still outstanding on this are:
> 
>     1. Should the slides use different colors?
> 
>     2. What name should we go with?

The colors could be a follow up if we get a designer's opinion on the matter. Shouldn't block this PR from landing.

The name, I'm not sure. I kinda like stacks, but it's again a subjective opinion.

**Review by MaggieCabrera — APPROVED** [review]

**Comment by scruffian** [comment]
Ok I'm going to merge this, but I think it would still be good to get more opinions on the name and colors. Happy to change them after the merge.

**Pull request merged**
